streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicConfig.java

```diff
@@ -20,44 +20,27 @@
 
 import java.util.Map;
 import java.util.Objects;
-import java.util.Properties;
-import java.util.Set;
 
 /**
  * InternalTopicConfig captures the properties required for configuring
  * the internal topics we create for change-logs and repartitioning etc.
  */
-public class InternalTopicConfig {
-    public enum CleanupPolicy { compact, delete }
+public abstract class InternalTopicConfig {
 
-    private final String name;
-    private int numberOfPartitions = -1;
-    private final Map<String, String> logConfig;
-    private final Set<CleanupPolicy> cleanupPolicies;
+    // we need to distinguish windowed and un-windowed store changelog since their cleanup policy may be different
+    public enum InternalTopicType { REPARTITION, WINDOWED_STORE_CHANGELOG, UNWINDOWED_STORE_CHANGELOG }
```

Comment:
I guess this can be removed now?

Reply:
Ack, will do in the docs PR.

Reply:
Just checked the code, it is indeed removed already